skyportal/handlers/api/instrument.py

```diff
@@ -39,7 +39,17 @@ async def post(self):
         self.verify_and_commit()
 
         if field_data is not None:
-            print(instrument.id)
+            if field_of_view_shape is None:
+                return self.error('field_of_view_shape is required with field_data')
+            elif field_of_view_shape in ['square', 'circle']:
+                if field_of_view_size is None:
+                    return self.error(
+                        f'field_of_view_size is required for field_of_view_shape={field_of_view_shape}'
+                    )
+            elif field_of_view_shape == "ZTF":
+                pass
+            else:
+                return self.error('field_of_view_shape must be square, circle, or ZTF')
```

Comment:
I'd replace this with an earlier call:

```
if field_of_view_shape not in ('square', 'circle', 'ZTF'):
    return self.error(...)
```

Reply:
So, maybe the whole block could be something like:

```
if field_of_view_shape is None:
    return self.error('`field_of_view_shape` is required with field_data')
if field_of_view_shape not in ('square', 'circle', 'ZTF'):
    return self.error('field_of_view_shape must be square, circle, or ZTF')
if field_of_view_size is None:
    return self.error(
        f'field_of_view_size is required for field_of_view_shape={field_of_view_shape}'
    )
```

Reply:
yes I think I got it fixed like this now.